astropy/io/ascii/latex.py

```diff
@@ -384,7 +384,12 @@ def start_line(self, lines):
 
     def write(self, lines):
         lines.append(self.data_start)
+        lines_length_initial = len(lines)
         core.BaseData.write(self, lines)
+        # To remove extra space and // appended which creates an extra new line
+        # in the end.
+        if len(lines) > lines_length_initial:
```

Comment:
I thought about that, too, when I reviewed it, and decided that the ways it's implemented now is better for exactly the case of an empty table you mention. I don't think that ever comes up, but we should not put in bug just for the sake of saving one line of code (and this is not performance critical - nobody writes millions of AASTeX tables).